### DISCUSSION TITLE
Extended refresh action to tables and repeaters

### DISCUSSION THREAD

**Discussion by diogenesbrussels** [Budibase/budibase#6111] [Ideas]
Hello,

Would it be possible to extend the refresh action to other components than the data provider? It would be useful to be able to refresh a repeater or a table after a query has been executed. I think this would allow more flexibility!

Thanks in advance!
Olivier

**Comment by melohagan** [comment]
Hi @diogenesbrussels 

If you refresh the data provider of a repeater or a table, then it will refresh them as well. Maybe there's a scenario I'm missing were this is not the case?

**Reply by diogenesbrussels** [reply]
Hi @melohagan,

Thanks for the quick feedback. Sorry, I was a bit confused 😕 about the exact component I am using. I'm actually talking about block components: 'repeater block', 'table block', etc.
I was using a block component and couldn't figure out how to refresh it. I can modify this to move forward but I would find it more logical to be able to refresh the block components as they contain the data provider directly. Maybe this is not necessary?

Thanks!

**Reply by melohagan** [reply]
Hi @diogenesbrussels 

Yes I understand you now! That is an unfortunate limitation of blocks. 

I think there are talks going on as how to get the most out of blocks, so that there is less of a difference between a block and the more flexible data provider/component combo. 

I must admit, I rarely use blocks because it does limit your options. Hopefully general block improvements will be made over the coming months!

**Reply by diogenesbrussels** [reply]
@melohagan thanks for the clarification, I will redo the relevant views with a data provider!
